GVFS/GVFS.Common/Prefetch/Jobs/ReadFilesJob.cs

```diff
@@ -38,8 +39,9 @@ protected override void DoWork()
                 string blobId;
                 while (this.availableBlobs.TryTake(out blobId, Timeout.Infinite))
                 {
-                    foreach (string path in this.blobIdToPaths[blobId])
-                    {
+                    foreach (Tuple<string, string> modeAndPath in this.blobIdToPaths[blobId])
+                    {
+                        string path = modeAndPath.Item2;
```

Comment:
With your own class type you could drop the `path` temporary variable and just pass in `modeAndPath.Path` to `HydrateFile`.